ts-tests/tests/test-eth-balance.ts

```diff
@@ -0,0 +1,170 @@
+import { ApiPromise, WsProvider, Keyring } from "@polkadot/api";
+import { KeyringPair } from '@polkadot/keyring/types';
+import { U8aFixed } from '@polkadot/types/codec';
+import * as crypto from '@polkadot/util-crypto';
+import { testValidator } from "@polkadot/util-crypto/base32/is";
+import { expect } from "chai";
+
+// Provider is set to localhost for development
+const wsProvider = new WsProvider("ws://localhost:9944");
+
+// Keyring needed to sign using Alice account
+const keyring = new Keyring({ type: 'sr25519' });
+
+// Configs of test ropsten account
+const test_eth_address = "[0x167453188a05082e3b347c1c518f3dd55d37fbbf]";
+const test_r = '0x4306864b7928f8393579ad849fd47685840cdf197c0085a5d241d28b9c6d14b3';
+const test_s = '0x3f4d2f5912213b3c0983c01a2199759b794d1423fe66cd5076b6ca9f5f60e45f';
+const test_v = 27;
+
+
+// Setup the API and Alice Account
+async function init() {
+	console.log(`Initiating the API (ignore message "Unable to resolve type B..." and "Unknown types found...")`);
+
+	// Initiate the polkadot API.
+	const api = await ApiPromise.create({
+		provider: wsProvider,
+		types: {
+			// mapping the actual specified address format
+			Address: "AccountId",
+			// mapping the lookup
+			LookupSource: "AccountId",
+			Account: {
+				nonce: "U256",
+				balance: "U256"
+			},
+			Transaction: {
+				nonce: "U256",
+				action: "String",
+				gas_price: "u64",
+				gas_limit: "u64",
+				value: "U256",
+				input: "Vec<u8>",
+				signature: "Signature"
+			},
+			Signature: {
+				v: "u64",
+				r: "H256",
+				s: "H256"
+			}
+		}
+  });
+
+	console.log(`Initialization done`);
+	console.log(`Genesis at block: ${api.genesisHash.toHex()}`);
+
+	const alice = keyring.addFromUri('//Alice', { name: 'Alice default' });
+
+	const { nonce, data: balance } = await api.query.system.account(alice.address);
+	console.log(`Alice Substrate Account: ${alice.address}`);
+	console.log(`Alice Substrate Account (nonce: ${nonce}) balance, free: ${balance.free.toHex()}`);
+
+	return { api, alice };
+}
+
+// Create Ethereum Link from ALICE
+async function eth_link(api: ApiPromise, alice: KeyringPair) {
+
+  console.log(`\nStep 1: Link Ethereum account`);
+
+  const transaction = api.tx.accountLinkerModule.link(alice.address, 0, 0, 0, test_r, test_s, test_v);
```

Comment:
Have you tested this? The interface seems be not correct. And expiring_block_number=0 seems will never succeed unless you use manual seal.

Reply:
As said, it's still with old account linker implementation. This will be fixed once it's merged